src/components/HomePage.js

```diff
@@ -0,0 +1,87 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { fetchData } from '../redux/weather/weathersReducer';
+import chooseImage from './ImageFunction';
+import './style/homePage.css';
+import microphone from '../Image/green-radio-microphone-14640.png';
+import setting from '../Image/settings-5666.png';
+import rightIcon from '../Image/right-arrow-6421.png';
+
+const ShowAll = () => {
+  const dispatch = useDispatch();
+  const weatherData = useSelector((state) => state.weathers.weatherData);
+  const [selectedOption, setSelectedOption] = useState('All'); // State to store the selected option value
+
+  useEffect(() => {
+    dispatch(fetchData());
+  }, [dispatch]);
+  const filteredData = weatherData.filter((city) => {
+    if (selectedOption === 'All') {
+      return true; // Show all cities when 'All' is selected
```

Comment:
- [ ] Great job adding a filter option on the page that filters temperatures but on the homepage, there should also be a search option where users can filter items by searching. In this project, users can search for countries. Please make sure to implement this.

![image](https://github.com/misikir21/Weather/assets/83803414/b2016fff-08cb-486e-8238-bbe7fbc5c6da)